shared/util/tf/tf_parser.go

```diff
@@ -190,24 +190,25 @@ func (tf TfParser) ParseModuleCall(happyProjectRoot, tfDirPath string) (ModuleCa
 				return errs.Wrap(diags.Errs()[0], "terraform code has errors")
 			}
 
-			if !strings.Contains(source.AsString(), "modules/happy-stack-") {
-				// Not a happy stack module
-				continue
-			}
-
-			tempDir, err := os.MkdirTemp("", "happy-stack-module")
-			if err != nil {
-				return errs.Wrap(err, "Unable to create temp directory")
-			}
-			defer os.RemoveAll(tempDir)
+			sourceDir := ""
+			if strings.Contains(source.AsString(), "modules/happy-stack-") {
+				sourceDir, err = os.MkdirTemp("", "happy-stack-module")
+				if err != nil {
+					return errs.Wrap(err, "Unable to create temp directory")
+				}
+				defer os.RemoveAll(sourceDir)
 
-			// Download the module source
-			err = getter.GetAny(tempDir, source.AsString())
-			if err != nil {
-				return fmt.Errorf("%w: %w", err, ErrUnableToDownloadModuleSource)
+				// Download the module source
+				err = getter.GetAny(sourceDir, source.AsString())
+				if err != nil {
+					return fmt.Errorf("%w: %w", err, ErrUnableToDownloadModuleSource)
+				}
+			} else {
+				// local reference
+				sourceDir = fmt.Sprintf("%s%s%s", happyProjectRoot, "/.happy/terraform/", source.AsString()[2:])
```

Comment:
This works for source set to `git@github.com:chanzuckerberg/happy//terraform/modules/happy-stack-eks?ref=main` and `./modules/ecs-stack`, but what about `./modules/happy-stack-eks`? Is it possible to determine whether the source is remote or local by looking at the source structure? Something like - if it starts with a period, it's local, if it starts with `git@` or `https:` or contains `github.com:` it's remote?

Reply:
Good call. I've changed the logic of checking local reference. I think all our remote reference starts with `git@`, but I could also check for `https:` if that would be more safe.

Reply:
I don't mean to drag it out :) Let's create a method that determines if a path is local, and use if for both tf_parser and hcl_manager. Also, let's check on to determine a remote source 

```
			if strings.HasPrefix(source.AsString(), "git@") || strings.HasPrefix(source.AsString(), "https:") {
```

Reply:
No worries! Let me added this and verify that it works!

Reply:
Move the logic into a shared function reside in `tf_generatator.go`, and added `http`

Reply:
I came across the terraform doc on this subject, it's more complex than I thought :) https://developer.hashicorp.com/terraform/language/modules/sources . Based on what we're dealing with, maybe, it's easier to determine the locally sourced modules. Per doc, it is 

```
A local path must begin with either ./ or ../ to indicate that a local path is intended, to distinguish from a module registry address.
```